CONTRIBUTING.md

```diff
@@ -317,6 +317,8 @@ GitHub provides reactions as a simple and quick way to provide feedback to pull
 
 Pull request reviews should include a list of what has been reviewed in a comment, so other reviewers and mergers can know the state of the review.
 
+Pull request reviews that only nitpick things that are easily fixed in treewides cause unnecessary friction and are better left unposted. Nitpicks may however accompany blockers.
```

Comment:
Another thing I've recently started doing is to simply PR your nitpicks against the contributor's branch or even just push them directly if you're a committer (most people leave the "allow maintainers to push commits" setting on in PRs).

I do find it hard to "outlaw" nitpicks like this however. Sometimes, nitpicks do have merit like asking for code comments explaining patches or unusual code. You could argue that those aren't nitpicks though I guess..

Reply:
I usually avoid pushing to contributor forks, and sending a PR to their fork usually results in a merge commit they have to clean up.

Most of the time I add code `suggestion`s, sometimes if it is only stylistic issues or some poor but benign use of `pname`, I let it slide and leave it for cleanup treewides. If I discover a unrelated issue while reviewing something, prefer not to stray from the scope of the PR and rather go and fix it in a separate PR.

Reply:
What I also prefer to do these days is approve the PR with a set of nitpicks. By doing this, I try to convey that I'd accept the PR as is but would prefer the nitpicks to be solved. Usually people fix them the next day and I merge but if they don't it should be merged as is a few days later. Perhaps we should establish that as a standard so that other committers can just merge when they see the latter case.

I wish we had some sort of automation that would queue a merge for a week from now if nothing changes. (If the contributor made changes in the mean time, I'd get pinged and can just merge manually.)

Reply:
> Another thing I've recently started doing is to simply PR my nitpicks against the contributor's branch or even just push them directly if you're a committer

+1, and I think that as long as you don't force-push it's not particularly intrusive

Reply:
For those interested, i've started proposing patches in comments ([example](https://github.com/NixOS/nixpkgs/pull/336023#issuecomment-2396823997)). This proved easy to script with `git diff --staged`, `$EDITOR` and `gh pr review --body-file`